apps/cmp.c

```diff
@@ -1837,64 +1842,126 @@ static int setup_request_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
     return 0;
 }
 
-static int handle_opt_geninfo(OSSL_CMP_CTX *ctx)
+static int add_certProfile(OSSL_CMP_CTX *ctx, const char *name)
 {
-    long value;
-    ASN1_OBJECT *type;
-    ASN1_INTEGER *aint;
-    ASN1_TYPE *val;
-    OSSL_CMP_ITAV *itav;
-    char *endstr;
-    char *valptr = strchr(opt_geninfo, ':');
+    OSSL_CMP_ITAV *itav = NULL;
+    STACK_OF(ASN1_UTF8STRING) *sk;
+    ASN1_UTF8STRING *utf8string;
 
-    if (valptr == NULL) {
-        CMP_err("missing ':' in -geninfo option");
+    if (ctx == NULL || name == NULL)
         return 0;
-    }
-    valptr[0] = '\0';
-    valptr++;
 
-    if (!CHECK_AND_SKIP_CASE_PREFIX(valptr, "int:")) {
-        CMP_err("missing 'int:' in -geninfo option");
+    if ((sk = sk_ASN1_UTF8STRING_new_reserve(NULL, 1)) == NULL)
         return 0;
-    }
+   if ((utf8string = ASN1_UTF8STRING_new()) == NULL)
+       goto err;
+   if (!ASN1_STRING_set(utf8string, name, (int)strlen(name))) {
+       ASN1_STRING_free(utf8string);
+       goto err;
+   }
+   (void)sk_ASN1_UTF8STRING_push(sk, utf8string); /* must succeed */
```

Comment:
Should this `OSSL_assert`, instead of ignoring the error?

Reply:
There is no `OSSL_assert`, so I presume you mean `ossl_assert`.
This is only available internally,
while in `apps/` there are many such uses of `(void)` where nothing can go wrong.
In this case there can be no OOM error due to `sk_ASN1_UTF8STRING_new_reserve(NULL, 1)`.